src/algorithms/preprocessing/reweighing.jl

```diff
@@ -0,0 +1,110 @@
+"""
+Helper function for ReweighingWrapper and ReweighingSamplingWrapper.
+grps is an array of values of protected attribute. y is an array of ground truth values.
+Array of (Frequency) weights are returned from this function.
+"""
+function _calculateWeights(grps, y)
+    grps_classes = levels(grps)
+    n_classes = length(grps_classes)
+    n = length(y)
+
+    fav = y.==levels(y)[2] # Boolean
+    unfav = y.==levels(y)[1]
+    n₊ = sum(fav) # Total Number of favourable outcomes
+
+    weights = zeros(Float64, n)
+    for i in 1:n_classes
+        class = grps.==grps_classes[i]
+        weights[class .& fav] .= n₊*sum(class)/(n*sum(class .& fav))
+        weights[class .& unfav] .= (n-n₊)*sum(class)/(n*sum(class .& unfav))
+    end
+    return weights
+end
+
+"""
+    ReweighingWrapper
+
+ReweighingWrapper is a preprocessing algorithm wrapper in which Weights for each group-label combination is calculated.
+These calculated weights are then passed to the classifier model which further uses it to make training fair.
+"""
+mutable struct ReweighingWrapper <: DeterministicNetwork
+    grp::Symbol
+    classifier::MLJBase.Model
+end
+
```

Comment:
replace `DeterministicNetwork` with `DeterministicComposite` (although the old name is aliased)